esphome/components/light/light_traits.h

```diff
@@ -10,33 +13,47 @@ class LightTraits {
 
   bool get_supports_brightness() const { return this->supports_brightness_; }
   void set_supports_brightness(bool supports_brightness) { this->supports_brightness_ = supports_brightness; }
-  bool get_supports_rgb() const { return this->supports_rgb_; }
-  void set_supports_rgb(bool supports_rgb) { this->supports_rgb_ = supports_rgb; }
-  bool get_supports_rgb_white_value() const { return this->supports_rgb_white_value_; }
-  void set_supports_rgb_white_value(bool supports_rgb_white_value) {
-    this->supports_rgb_white_value_ = supports_rgb_white_value;
+
+  std::set<ColorMode> get_supported_color_modes() const { return this->supported_color_modes_; }
```

Comment:
```suggestion
  const std::set<ColorMode> &get_supported_color_modes() const { return this->supported_color_modes_; }
```

Prevents a copy of the set if I'm not mistaken